app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php

```diff
@@ -124,32 +122,64 @@ public function draw()
                 $feed = 35;
                 $name = $childItem->getName();
             }
-            $line[] = ['text' => $this->string->split($name, 35, true, true), 'feed' => $feed];
+            $lines[$index][] = ['text' => $this->string->split($name, 35, true, true), 'feed' => $feed];
 
             // draw SKUs
             if (!$childItem->getOrderItem()->getParentItem()) {
                 $text = [];
                 foreach ($this->string->split($item->getSku(), 17) as $part) {
                     $text[] = $part;
                 }
-                $line[] = ['text' => $text, 'feed' => 255];
+                $lines[$index][] = ['text' => $text, 'feed' => 255];
+                $index++;
             }
 
             // draw prices
             if ($this->canShowPriceInfo($childItem)) {
-                $price = $order->formatPriceTxt($childItem->getPrice());
-                $line[] = ['text' => $price, 'feed' => 395, 'font' => 'bold', 'align' => 'right'];
-                $line[] = ['text' => $childItem->getQty() * 1, 'feed' => 435, 'font' => 'bold'];
+                $lines[$index][] = ['text' => $childItem->getQty() * 1, 'feed' => 435, 'align' => 'right'];
 
                 $tax = $order->formatPriceTxt($childItem->getTaxAmount());
-                $line[] = ['text' => $tax, 'feed' => 495, 'font' => 'bold', 'align' => 'right'];
-
-                $row_total = $order->formatPriceTxt($childItem->getRowTotal());
-                $line[] = ['text' => $row_total, 'feed' => 565, 'font' => 'bold', 'align' => 'right'];
+                $lines[$index][] = ['text' => $tax, 'feed' => 495, 'font' => 'bold', 'align' => 'right'];
+
+                $this->_item = $childItem;
+                $feedPrice = 380;
+                $feedSubtotal = $feedPrice + 185;
+                foreach ($this->getItemPricesForDisplay() as $priceData) {
```

Comment:
Could you please try to refactor this piece? This method already has ExcessiveMethodLength suppress.

Reply:
refactored draw method